app/services/streaming/streaming.ts

```diff
@@ -792,45 +824,78 @@ export class StreamingService
 
     this.powerSaveId = remote.powerSaveBlocker.start('prevent-display-sleep');
 
-    // start streaming
+    // handle start streaming and recording
     if (this.views.isDualOutputMode) {
       // start dual output
+      if (this.dualOutputService.views.recordVertical) {
+        // stream horizontal and record vertical
+        const horizontalContext = this.videoSettingsService.contexts.horizontal;
+        NodeObs.OBS_service_setVideoInfo(horizontalContext, 'horizontal');
 
-      const horizontalContext = this.videoSettingsService.contexts.horizontal;
-      const verticalContext = this.videoSettingsService.contexts.vertical;
+        const signalChanged = this.signalInfoChanged.subscribe(
+          async (signalInfo: IOBSOutputSignalInfo) => {
+            if (signalInfo.service === 'default') {
+              if (signalInfo.signal === EOBSOutputSignal.Start) {
+                // Refactor when migrate to new API, sleep to allow state to update
+                await new Promise(resolve => setTimeout(resolve, 300));
+                this.toggleRecording();
+                signalChanged.unsubscribe();
```

Comment:
If I remember correctly RxJS didn't play nicely when unsubscribing to a subject inside a `subscribe` function. I believe this could be achieved with `take(1)` or `takeWhile(condition)` and then `subscribe` which would automatically unsubscribe for us. 
Bear in mind is been a while so API might have changed or even this assumption. 
```
this.signalInfoChanged.pipe(
  take(1)
 ).subscribe(...)
 ```
I believe the filter (if takeWhile isn't needed after all, it might) and the delay could also be achieved with operators, but that's not a real concern, just the subscribe/unsubscribe behavior is.

Reply:
Just for tracking, discussed